dan_layer/template_lib/src/models/metadata.rs

```diff
@@ -58,3 +60,13 @@ impl Default for Metadata {
         Self::new()
     }
 }
+
+impl Display for Metadata {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut s = vec![];
+        for (key, value) in self.0.iter() {
+            s.push(format!("key = {}, value = {}", key, value))
+        }
+        write!(f, "Metadata: {:?}", s)
```

Comment:
Generally, the implementation of Display should never allocate
```suggestion
        write!(f, "Metadata: ")
        for (key, value) in self.0.iter() {
            write!!("key = {}, value = {} ", key, value))
        }

```

Reply:
Thanks, addressed !